{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "46488cec-5180-44a2-8d3a-16c05c5d0d6b",
   "metadata": {},
   "source": [
    "### Download the Test data from MaRS bioproject (https://www.ncbi.nlm.nih.gov/bioproject/PRJNA428490)\n",
    "\n",
    "User can download five samples from MaRS bioproject (SRR18918405, SRR18918415, SRR18918416, SRR18918242, SRR18918253) using SRA Toolkit, rename the samples to AMD IDs using the code provided below and run the workflow.  \n",
    "\n",
    "#### Download SRA toolkit from https://trace.ncbi.nlm.nih.gov/Traces/sra/sra.cgi?view=software\n",
    "    \n",
    "#### If using Conda to download SRA toolkit, Run\n",
    "\n",
    "`    \n",
    "conda install -c bioconda sra-tools\n",
    "`  \n",
    "    \n",
    "#### After user download and Install the tool, Run the commands below to download the fastq files.\n",
    " \n",
    "* prefetch will download and save SRA file related to SRR accession in the current directory \n",
    "  \n",
    "\n",
    "`\n",
    "prefetch SRR18918405 SRR18918415 SRR18918416 SRR18918242 SRR18918253\n",
    "`\n",
    "\n",
    "* convert to FASTQ: fastq-dump will convert sra to fastq, use --split-files option for paired data\n",
    "\n",
    "`\n",
    "fastq-dump SRR18918405 SRR18918415 SRR18918416 SRR18918242 SRR18918253 --split-files\n",
    "`\n",
    "\n",
    "`mkdir SRA_Testdata\n",
    "`\n",
    "\n",
    "`\n",
    "mv *.fastq SRA_Testdata\n",
    "`   \n",
    "    \n",
    "Now All the fastq Paired end reads will be saved in folder SRA_Testdata\n",
    "\n",
    "\n",
    "   "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "d7624f99-d724-4a6c-98ed-2be1356ea85b",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "id": "9b9fe5a1-1838-4a7d-95e2-6fdbc36b9de6",
   "metadata": {},
   "source": [
    "## Rename sample ID with AMD ID\n",
    "\n",
    "## Background: \n",
    "\n",
    "   CDC Malaria Next generation Sequancing lab generates sample ID which is called AMD ID, is 20 character long and have specific meta data attched to it which makes it easier to standerdize the sample name and tracking. It also makes it convenient to submit the samples to NCBI as it requires to submit associated meta data (attributes)submitted with each fastq record.\n",
    "  \n",
    "   **AMD ID** is standerdized sample identification number which includes ID number and associated metadata related to the sample. \n",
    " \n",
    "   - AMD ID and bit code key is found under MS Teams > SOPs - Lab > Files > Sample Naming > Sample_naming_key.pptx <pre>\n",
    "   \n",
    "   \n",
    "   - Key: **Year Country State/Site DayofTreatment Treatment SampleID Genus SampleType GeneMarker-8bitcode SampleSequencingCount**\n",
    "\n",
    "   \n",
    "        - Example:\n",
    "    \n",
    "            - Individual sequenced sample ID: 17GNDo00F0001PfF1290 = 2017 Guinea Dorota Day0 AS+AQ 0001 P.falciparum FilterBloodSpot k13-crt-mdr-dhfr-dhps-cytB-cpmp-pfs47 0 \n",
    "                          \n",
    "            - Pooled sequenced sample ID: 17GNDo00X001p10F1290 = 2017 Guinea Dorota Day0 **X** 001**Pooled SamplesInPool** FilterBloodSpot k13-crt-mdr-dhfr-dhps-cytB-cpmp-pfs47 0  \n",
    "           \n",
    "              - NOTE: If information is not availble (na) **x** is used. For pooled samples, Treatment is x since its a pool of multiple samples with that info.\n",
    "          \n",
    "              - NOTE: For pooled samples, **Sample ID** is replaced with **three digit sample ID + p(Pooled)** and **Genus** is replaced **total number of SamplesInPool** to indicated this as a pooled sequenced sample and sample count in each pool. \n",
    "     \n",
    "## Code information   \n",
    "  **Samples_ids.csv** files contains SRR ids and AMD ids. This code runs on a fastq files downloaded from MaRS bioproject SRA and  renmane it with AMD ids provided in sample_ids file. It uses OS module from python, renmae each fastq file with newname. \n",
    "\n",
    "### Required packages\n",
    "- [Pandas](https://pandas.pydata.org/) \n",
    "-  os \n",
    "\n",
    "### Inputs\n",
    "- A folder containig Fastq_files from SRA, .csv file with SRR name and AMD ID.\n",
    "\n",
    "### Outputs:\n",
    "- rename a files with standerdized AMD IDs.\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "5a71be27-67bd-4a54-ad72-cf4df92f19bd",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import os\n",
    "\n",
    "\n",
    "\n",
    "My_files =  \"/Users/dhruvipatel/Desktop/Archive/Training_senegal/Test/SRA_Testdata\"   # folder path contains fastq files\n",
    "\n",
    "\n",
    "with open('Samples_ids.csv') as csv_file:    #open a csv file\n",
    "    next(csv_file)                           # remove a header \n",
    "    for line in csv_file:                    # read a file line by line\n",
    "        line = line.split(\",\")\n",
    "        field_1 = line[0]                    # field_1 contains column1 (SRR_ID) data of .csv file\n",
    "        field_2 = line[1].strip()            # field_2 contains column2 (AMD_ID) data of .csv file\n",
    "        \n",
    "        for file in os.listdir(My_files):                               \n",
    "            SRR_ID = file.split(\"_\")[0]                           # for ids in old files, split it at underscore(_). only get first part of ID before underscore(_).\n",
    "            AMD_ID = \"_\".join(file.split(\"_\")[1:])              # for ids in old files, split it at underscore(_). remove first part of ID before _ and join again using underscore\n",
    "            print(SRR_ID, AMD_ID)\n",
    "            if SRR_ID in field_1 :\n",
    "                \n",
    "                oldname = file\n",
    "                newname =  field_2 + \"_\"+ AMD_ID\n",
    "                old_path = os.path.join(My_files, oldname)                                           # OLD path of samples\n",
    "                print(\"OLD \", old_path)  \n",
    "                new_path = os.path.join(My_files, newname)                                           # Path for new IDs\n",
    "                print(\"NEW \", new_path) \n",
    "\n",
    "            \n",
    "                os.rename(old_path, new_path) \n",
    "            \n",
    "            \n",
    "            \n",
    "            "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "6b21df84-6d63-46dd-b5f2-3fa788b08ea3",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.12"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
